### PR TITLE
[NEW] Add 'unread messages' marker

### LINKED ISSUE #667: [NEW] Support "unread" markup in the list of messages
The list of messages need to show a "Unread" mark when the messages below weren't read by the user yet.

### PR DESCRIPTION
**Pull request by shriakhilc**
@RocketChat/android

Closes #667

I've modeled the marker to be similar to the one displayed in the browser. I've noticed two other issues while fixing this one:
* The Room's `ls` value is not updated when visited via the app. This leads to a number of UX issues. I believe [this](https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/issues/49) is the corresponding issue in the Kotlin SDK, and it seems to have been merged recently. I would like to try implementing this on the app side.

* The `ls` field is marked `lm` in the `Subscriptions.kt` file [here](https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/blob/99b4ce6c2e79a7c6f7a4d2a9e3a8b2b719b94720/core/src/main/kotlin/chat/rocket/core/internal/model/Subscription.kt#L22). I modified this on my local repo for testing, and without it the feature will not work properly since the server is sending the `ls` field.

Should I open separate Issues for both of these?

A screenshot of the marker in action:
screenshot_20180216-030946

### CODE REVIEW

File: app/src/main/res/layout/item_message.xml
Comment (rafaelks) [inline app/src/main/res/layout/item_message.xml]: I believe we should use i18n strings here?

**Comment by rafaelks** [comment]
@TheGamer007 Thank you so much for your PR! We'll be doing some review in your code soon. 👍

File: app/src/main/res/layout/item_message.xml
Comment (shriakhilc) [reply app/src/main/res/layout/item_message.xml]: I'll add it to the next commit, together with any other review issues.

Comment (philipbrito) [inline app/src/main/res/layout/item_message.xml]: Add this string to the [strings.xml](https://github.com/RocketChat/Rocket.Chat.Android/blob/develop-2.x/app/src/main/res/values-pt-rBR/strings.xml) file.

You can also add a new entry for this string in [values-pt-rBR/strings.xml](https://github.com/RocketChat/Rocket.Chat.Android/blob/develop-2.x/app/src/main/res/values-pt-rBR/strings.xml) with the value `Mensagens não lidas`  :)

`<string name="msg_unread_messages">...</string>`

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by shriakhilc** [comment]
@filipedelimabrito changes made. Thanks for the translation!

**Comment by shriakhilc** [comment]
@filipedelimabrito is there something more to be done, or are we just waiting for the others to review?

**Comment by luciofm** [comment]
@TheGamer007 will do a review later today...

**Comment by shriakhilc** [comment]
Rebased onto the latest changes

**Comment by luciofm** [comment]
@TheGamer007 I'm working on some breaking changes, that will conflict with this PR. Once I'm done, I'll work with you to update it and merge

**Comment by shriakhilc** [comment]
@luciofm Any updates on your changes?

**Comment by rafaelks** [comment]
@TheGamer007 Thanks for your PR! Could you please resolve the conflicts of the PR?

**Comment by shriakhilc** [comment]
@rafaelks Sure. I was waiting for @luciofm to finish merging his breaking changes before fixing the conflicts. I'll rebase and push in a while

**Comment by shriakhilc** [comment]
@rafaelks Conflicts resolved

**Comment by shriakhilc** [comment]
@luciofm I've resolved the conflicts. Let me know if any other fixes are needed before this can be merged.

**Pull request merged**
